Content.Server/Medical/HealthAnalyzerSystem.cs

```diff
@@ -52,6 +55,10 @@ private void OnDoAfter(EntityUid uid, HealthAnalyzerComponent component, DoAfter
             _audio.PlayPvs(component.ScanningEndSound, args.Args.User);
 
             UpdateScannedUser(uid, args.Args.User, args.Args.Target.Value, component);
+
+            TryComp<PowerCellDrawComponent>(uid, out var draw);
+            if (!_powerCell.HasActivatableCharge(uid, battery: draw))
+                _appearance.SetData(uid, PowerCellSlotVisuals.Enabled, false);
```

Comment:
This should be in the TryUseActivatableCharge function.

Reply:
I don't understand what exactly I'm doing wrong, since now it's not working after porting this over to the TryUseActivatableCharge function. Posted it on Discord in howdoicode